### PR TITLE
[TestFix] Fix gluster-block volume name issue

### PR DESCRIPTION
**Pull request by SushilG96**
Gluster block volume deletion gets fail when heketi volume ID with a prefix is used as gluster-block volume name in case custom name is used while creating heketi volume. This PR will fix the issue by generating a gluster-block name using the heketi volume name.

Signed-off-by: SushilG96 <susgupta@redhat.com>

### CODE REVIEW

**Review by vamahaja — APPROVED** [review]

File: tests/functional/heketi/test_block_volumes_heketi.py [hunk 1]
```diff
@@ -371,25 +371,27 @@ def test_create_max_num_blockhostingvolumes(self):
...
+                bhv_id, json=True)
             self.assertNotIn(
                 "blockvolume", heketi_vol_info["blockinfo"].keys())
             gluster_vol_info = get_block_list(
-                'auto_get_gluster_endpoint', volname="vol_%s" % bhv_volume)
+                'auto_get_gluster_endpoint', bhv_name)
             self.assertIsNotNone(
-                gluster_vol_info, "Failed to get volume info %s" % bhv_volume)
-            new_bhv_list.remove(bhv_volume)
+                gluster_vol_info, "Failed to get volume info %s" % bhv_name)
```
Comment (adityaramteke) [inline tests/functional/heketi/test_block_volumes_heketi.py:383]:
```suggestion
                gluster_vol_info, "Failed to get volume info {}".format(bhv_name))
```

File: tests/functional/heketi/test_block_volumes_heketi.py [hunk 2]
```diff
@@ -322,7 +322,7 @@ def test_create_max_num_blockhostingvolumes(self):
                 bhv_name, default_bhv_size,
                 raise_on_cleanup_error=False, json=True, block=True)
             if bhv_info["id"] not in existing_bhv_list.keys():
-                new_bhv_list.append(bhv_info["id"])
+                new_bhv_list.append((bhv_info["id"], bhv_info["name"]))
```
Comment (adityaramteke) [inline tests/functional/heketi/test_block_volumes_heketi.py:325]: Can we use .get() instead?

**Review by adityaramteke — CHANGES_REQUESTED** [review]

File: tests/functional/heketi/test_block_volumes_heketi.py (hunk 1 above)
Comment (SushilG96) [reply tests/functional/heketi/test_block_volumes_heketi.py:383]: These are old changes, which this file follows if we want to change those, so we need to change the remaining.
So Continuing with the old string format

Comment (adityaramteke) [reply tests/functional/heketi/test_block_volumes_heketi.py:383]: So does that mean we should add the new change in the same format if we want to add a new string?

Comment (vamahaja) [reply tests/functional/heketi/test_block_volumes_heketi.py:383]: @adityaramteke check commit messsage, this change is not for fixing old string format. It's for fixing known bug.

Comment (srivickynesh) [reply tests/functional/heketi/test_block_volumes_heketi.py:383]: @adityaramteke +1 to check commit msg

Comment (adityaramteke) [reply tests/functional/heketi/test_block_volumes_heketi.py:383]: Okay, resolving this threading. Other changes are LGTM.

**Review by adityaramteke — APPROVED** [review]

**Review by srivickynesh — APPROVED** [review]

**Pull request merged**